src/lib.rs

```diff
@@ -1427,6 +1594,38 @@ impl Url {
     /// If this URL is cannot-be-a-base, do nothing and return `Err`.
     ///
     /// Compared to `Url::set_host`, this skips the host parser.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use url::{Url, ParseError};
+    ///
+    /// # fn run() -> Result<(), ParseError> {
+    /// let mut url = Url::parse("http://example.com")?;
+    /// url.set_ip_host("127.0.0.1".parse().unwrap());
```

Comment:
I think it's acceptable because 127.0.0.1 is valid static ip address. But it will be much better if change return type to `Result<(), Box<Error>>`.

```
fn run() -> Result<(), Box<Error>> {
  url.set_ip_host("127.0.0.1".parse()?);
}
```